File: 3658_GCD_of_Odd_and_Even_Sums/solution.c
// Method -> 1
// int GCD(int a, int b) {
//     if (b == 0) {
//         return a;
//     }
//     return GCD(b, a % b);
// }

// int gcdOfOddEvenSums(int n) {
//     int odd = 0, even = 0;
//     for (int i = 1; i <= n; i++) {
//         even += i * 2;
//         odd += (i * 2 - 1);
//     }

//     return GCD(odd, even);
// }

// Method -> 2
// int GCD(int a, int b) {
//     if (b == 0) {
//         return a;
//     }
//     return GCD(b, a % b);
// }

// int gcdOfOddEvenSums(int n) {
//     int odd = n * n;
//     int even = n * (n + 1);

//     return GCD(odd, even);
// }

// Method -> 3 (Best Solution)
int gcdOfOddEvenSums(int n) { return n; }

File: 3658_GCD_of_Odd_and_Even_Sums/solution.py
# Method -> 1
# class Solution:
#     def gcd(a: int, b: int):
#         if b == 0:
#             return a
#         return gcd(b, a % b)

#     def gcdOfOddEvenSums(self, n: int) -> int:
#         even, odd = 0, 0
#         for i in range(n):
#             even += i * 2
#             odd += i * 2 - 1

#         return gcd(odd, even)

# Method -> 2
# class Solution:
#     def gcd(a: int, b: int):
#         if b == 0:
#             return a
#         return gcd(b, a % b)

#     def gcdOfOddEvenSums(self, n: int) -> int:
#         even = n * (n - 1)
#         odd = n * n

#         return gcd(odd, even)

# Method -> 3 (Best Solution)
class Solution:
    def gcdOfOddEvenSums(self, n: int) -> int:
        return n